### ISSUE TITLE
smoothxg throws   what():  basic_string::_M_construct null not valid

### ISSUE THREAD

**Issue by cgroza** [pangenome/smoothxg#16]
Hi,
I am running smoothxg on GFA graphs produced via vg construct > vg view.
```
cgroza@blg9122:~/.../sv-graph/graphs $ ~/smoothxg/bin/smoothxg -g chr1.vg.gfa
terminate called after throwing an instance of 'std::logic_error'
  what():  basic_string::_M_construct null not valid
Aborted (core dumped)
cgroza@blg9122:~/.../sv-graph/graphs $
```
This is the error I am getting.
I enabled debugging with -d and the output stays the same.
What could be causing this error and how could I fix it?

Cristian Groza

**Comment by ekg** [comment]
I've never tested on a graph built by vg construct. But this is a good use case, because you might be able to normalize the representation of complex variation in the VCF.

It's possible that you need to have paths covering every base in the graph for smoothing to not break like this.

Please try running `vg construct -a` when building your graph. Does that work?

**Comment by cgroza** [comment]
Hi,
Thanks for the quick reply.
I already passed the -a flag to vg construct and the paths are indeed saved in the vg file before I convert to GFA with vg view.
```
cgroza@beluga4:~/scratch/sv-graph $ vg paths -L -v graphs/chr1.vg | head
_alt_690f8d2fa0e64915865faf22fb22b9a320da3fd2_3
_alt_646ca4e078f643e66c6f63af905a593f2340c187_0
_alt_d9a3a44aafbdb32476a084ea5912eb2042a419a4_5
_alt_460556e6579145a11e1258b8a8ea4fad75b2eac7_10
_alt_ec650cad5a9dcf3f96ab1f7726ad150119570c29_1
_alt_8f499a29d6265f1c936e58362b92b8bf5605f744_0
_alt_1a4af475581ce7bf25bfb6d6b0079e7f4eb96d09_2
_alt_cdeaef57e2f1a0ce2887a7b68321614e862a1cd8_15
_alt_ab71928db30ae5ffdabc5e86b7f88844d4e5eed3_0
_alt_cdeaef57e2f1a0ce2887a7b68321614e862a1cd8_3
cgroza@beluga4:~/scratch/sv-graph $
```
So unfortunately, it does not work.
How would the path information be kept in the GFA file after conversion?

Cristian

**Comment by ekg** [comment]
The paths are under P lines in the GFA.

You don't get any output at all from smoothxg? By default, it provides
logging.

You could try passing the -n parameter, which will avoid some prep sorting
steps at the beginning that aren't necessary for VCF based graphs.

**Comment by cgroza** [comment]
So my GFA lines do indeed have P lines.
No, I only get 3 lines of output on stderr/stdout and they are the logic_error exception and core dump  above.
I passed the -n parameter and got equal results.

Cristian

**Comment by ekg** [comment]
What git commit are you running here? How did you build and install smoothxg?

Could you reproduce the same thing but on a very small graph that you can share?

**Comment by cgroza** [comment]
Yes I will try to adapt the test case in vg/test/small/x.fa and post it here.

**Comment by cgroza** [comment]
Hi,

Here is the minimal test case GFA. It has two Alu insertions right in the middle.
It was built from these files with vg construct -a > vg view.
[x.fa.gz](https://github.com/pangenome/smoothxg/files/5318519/x.fa.gz)
[small.vcf.gz](https://github.com/pangenome/smoothxg/files/5318520/small.vcf.gz)


[test.gfa.gz](https://github.com/pangenome/smoothxg/files/5318499/test.gfa.gz)

**Comment by ekg** [comment]
I've found the problem.

One of your P lines doesn't have the right number of fields. It's missing the path description.

```
-> % grep _alt_5d2a3c27da7879b7e1fb081ed7596ea49fe65e13_0 test.gfa
P       _alt_5d2a3c27da7879b7e1fb081ed7596ea49fe65e13_0
```

```
grep -v _alt_5d2a3c27da7879b7e1fb081ed7596ea49fe65e13_0 test.gfa >test1.gfa
smoothxg -g test1.gfa >test1.smooth.gfa
```

This should probably be checked somewhere in the xg process that's failing.

What is this field meant to represent? A deletion allele?

**Comment by ekg** [comment]
I suspect it's something made by vg construct with the -a flag.

**Comment by cgroza** [comment]
I think `vg construct -a` is adding these `_alt_...._0` every insertion site. Traversing this path would give you the reference sequence. So vg is confusing the reference path with a deletion allele at insertion sites?
Is this new behaviour in vg construct? I don't remember this behaviour in past versions.

**Comment by cgroza** [comment]
I removed the offending path and this was the output of smoothxg:

[smoothed.gfa.gz](https://github.com/pangenome/smoothxg/files/5355746/smoothed.gfa.gz)
